vars/sectionDeployToAKS.groovy

```diff
@@ -111,8 +121,11 @@ def call(params) {
                       timeoutWithMsg(time: config.fullFunctionalTestTimeout, unit: 'MINUTES', action: 'Functional tests') {
                         try {
                           builder.fullFunctionalTest()
-                        } finally {
                           savePodsLogs(dockerImage, params, "full-functional")
```

Comment:
Isn't this duplicate?

Reply:
Not sure what you mean there.

Before this change, when I added the pods logs, I added them in the `finally` clause as I wanted to have those logs regardless.
I still want that of course, but I can't use the `finally` block any longer as if I were to delete the helm release as part of the `catch` block, by the time I got to the `finally` the logs would be gone.

As a result I had to get rid of the `finally` block and explicitly put `savePodsLogs` both in the `try` and `catch` blocks, with the caveat being that the one in the `catch` block is executed BEFORE I clear the helm release.

Does it make sense or am I misunderstanding your concern?

Reply:
i don't think we want to save them twice. 

best to set a boolean value that tells the finally block the status

https://stackoverflow.com/a/10736283/4951015

```groovy
success = true
try {
 ...
} catch(err) {
  success = false
} finally {
  savePodLogs(...)
  if (!success) { 
    clearHelmReleaseForFailure(config, dockerImage, params, pcr)
  }
}
```

Reply:
I'd like to stress that we won't save them twice.

Anyway: in the interest of speed and since I don't own this repository, I'll follow your pattern.